### PR TITLE
Add File IO instrumentation docs

### PR DESCRIPTION
**Pull request by marandaneto**
Relates to https://github.com/getsentry/sentry-dart/pull/1160

### CODE REVIEW

File: src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 1]
```diff
@@ -31,3 +31,15 @@ The span finishes once the request has been executed. The span `status` depends
 When the HTTP request throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and `SentryEvent` will be linked when viewing it on the **Issue Details** page in [sentry.io](https://sentry.io).
 
 Learn more in our [Dio integration documentation](/platforms/dart/configuration/integrations/dio/#performance-monitoring-for-http-requests).
+
+### File I/O Instrumentation
+
+The Sentry-specific file I/O implementation starts a span out of the active span, bound to the scope for each file I/O operation. The SDK sets the span `operation` to `file.copy`/`file.write`/`file.delete`/`file.open`/`file.read`/`file.rename` and `description` to `filename`, e.g. `file.txt`
```
Comment (marandaneto) [inline src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx:37]: @AbhiPrasad what do you think about all these operations? I still can change it but Dart has specific methods per operation and if I don't do the operation suffix, everything would be `write/read` and you can't really tell which operation was executed.

Comment (marandaneto) [reply src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx:37]: cc @romtsn

Comment (AbhiPrasad) [reply src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx:37]: I think these operations are fine.

**Review by krystofwoldrich — APPROVED** [review]
LGTM.

File: src/platforms/dart/configuration/integrations/file.mdx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+description: "Learn more about the Sentry file I/O integration for the Dart SDK."
+---
+
+<Note>
+
+This feature is currently in Beta. Beta features are still in-progress and may have bugs. We recognize the irony.
+
+</Note>
+
+The `sentry_file` library provides [File](https://api.dart.dev/stable/2.18.5/dart-io/File-class.html) support for Sentry wrapping the `File` class with a `SentryFile` wrapper. It is able to run performance monitoring for `File` operations.
```
Comment (imatwawana) [inline src/platforms/dart/configuration/integrations/file.mdx:13]: This first sentence is a bit awkward. Suggest:
```suggestion
The `sentry_file` library provides [File](https://api.dart.dev/stable/2.18.5/dart-io/File-class.html) support which allows Sentry to wrap the `File` class with a `SentryFile` wrapper. It is able to run performance monitoring for `File` operations.
```

Also, what is "it" in the second sentence? The library? The SDK?

File: src/platforms/flutter/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 3]
```diff
@@ -213,3 +213,15 @@ The `SentryAssetBundle` instrumentation starts a span out of the active transact
 The `enableStructuredDataTracing` feature is available in Beta. Beta features are still in-progress and may have bugs. We recognize the irony.
 
 </Note>
+
+### File I/O Instrumentation
+
+The Sentry-specific file I/O implementation starts a span out of the active span, bound to the scope for each file I/O operation. The SDK sets the span `operation` to `file.copy`/`file.write`/`file.delete`/`file.open`/`file.read`/`file.rename` and `description` to `filename`, e.g. `file.txt`
```
Comment (imatwawana) [inline src/platforms/flutter/common/performance/instrumentation/automatic-instrumentation.mdx:219]:
```suggestion
The Sentry-specific file I/O implementation starts a span out of the active span, bound to the scope for each file I/O operation. The SDK sets the span `operation` to `file.copy`/`file.write`/`file.delete`/`file.open`/`file.read`/`file.rename` and `description` to `filename` (for example, `file.txt`).
```

File: src/platforms/flutter/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 4]
```diff
@@ -213,3 +213,15 @@ The `SentryAssetBundle` instrumentation starts a span out of the active transact
...
+
+In addition, the span contains other useful information such as `file.size` (raw number of bytes), `file.path` (an absolute path to the file) and `file.async` (`true` if the called method returns a `Future` or `false` if it's a `Sync` call) as part of the `data` payload.
```
Comment (imatwawana) [inline src/platforms/flutter/common/performance/instrumentation/automatic-instrumentation.mdx:221]:
```suggestion
In addition, the span contains other useful information such as `file.size` (raw number of bytes), `file.path` (an absolute path to the file) and `file.async` (`true` if the called method returns a `Future`, or `false` if it's a `Sync` call) as part of the `data` payload.
```

File: src/platforms/flutter/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 5]
```diff
@@ -213,3 +213,15 @@ The `SentryAssetBundle` instrumentation starts a span out of the active transact
...
+
+The span finishes once the operation has been executed. The span `status` is set to `SpanStatus.ok` if successful or `SpanStatus.internalError` if there was any error.
+
+When the operation throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the **Issue Details** page in sentry.io.
```
Comment (imatwawana) [inline src/platforms/flutter/common/performance/instrumentation/automatic-instrumentation.mdx:225]:
```suggestion
When the operation throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the **Issue Details** page in [sentry.io](https://sentry.io).
```

File: src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 6]
```diff
@@ -31,3 +31,15 @@ The span finishes once the request has been executed. The span `status` depends
...
+
+In addition, the span contains other useful information such as `file.size` (raw number of bytes), `file.path` (an absolute path to the file) and `file.async` (`true` if the called method returns a `Future` or `false` if it's a `Sync` call) as part of the `data` payload.
+
+The span finishes once the operation has been executed. The span `status` is set to `SpanStatus.ok` if successful or `SpanStatus.internalError` if there was any error.
+
+When the operation throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the **Issue Details** page in sentry.io.
```
Comment (imatwawana) [inline src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx:43]:
```suggestion
When the operation throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the **Issue Details** page in [sentry.io](https://sentry.io).
```

Comment (imatwawana) [inline src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx:43]: Should "SentryEvent" be in code font?
```suggestion
When the operation throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and `SentryEvent` will be linked when viewing it on the **Issue Details** page in sentry.io.
```

File: src/platforms/flutter/common/performance/instrumentation/automatic-instrumentation.mdx (hunk 5 above)
Comment (imatwawana) [inline src/platforms/flutter/common/performance/instrumentation/automatic-instrumentation.mdx:225]: Should "SentryEvent" be in code font?
```suggestion
When the operation throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and `SentryEvent` will be linked when viewing it on the **Issue Details** page in sentry.io.
```

File: src/platforms/dart/configuration/integrations/file.mdx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+title: File I/O
+sidebar_order: 5
+description: "Learn more about the Sentry file I/O integration for the Dart SDK."
+---
+
+<Note>
+
+This feature is currently in Beta. Beta features are still in-progress and may have bugs. We recognize the irony.
+
+</Note>
```
[the remark this answers is not in the document]
Comment (marandaneto) [reply src/platforms/dart/configuration/integrations/file.mdx:11]: TIL :)

File: src/platforms/dart/configuration/integrations/file.mdx (hunk 2 above)
Comment (marandaneto) [reply src/platforms/dart/configuration/integrations/file.mdx:13]: Yes, the `sentry_file` library that contains the wrapper

File: src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx (hunk 6 above)
Comment (marandaneto) [reply src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx]: Yep, can be.

File: src/platforms/dart/configuration/integrations/file.mdx (hunk 2 above)
Comment (marandaneto) [reply src/platforms/dart/configuration/integrations/file.mdx]: Changed it to:

> The `SentryFile` is able to run performance monitoring for `File` operations.

**Comment by marandaneto** [comment]
@imatwawana applied suggestions, thanks :)

**Review by imatwawana — APPROVED** [review]

File: src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx (hunk 1 above)
Comment (romtsn) [reply src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx]: @marandaneto could you also update this page please? https://develop.sentry.dev/sdk/performance/span-operations/#mobile

**Pull request merged**